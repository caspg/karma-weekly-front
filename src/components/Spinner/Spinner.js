import React from 'react';
import PropTypes from 'prop-types';

Spinner.propTypes = {
  mainColor: PropTypes.string,
  bgColor: PropTypes.string,
  radius: PropTypes.string,
  borderWidth: PropTypes.string,
};

Spinner.defaultProps = {
  mainColor: '#1779ba',
  bgColor: '#ddd',
  radius: '10em',
  borderWidth: '0.25em',
};

function Spinner(props) {
  const bgBorderStyle = `solid ${props.bgColor}`;
  const mainBorderStyle = `solid ${props.mainColor}`;

  const dynamicStyles = {
    borderLeft: mainBorderStyle,
    borderTop: mainBorderStyle,
    borderRight: bgBorderStyle,
    borderBottom: bgBorderStyle,
    borderWidth: props.borderWidth,
    width: props.radius,
    height: props.radius,
  };

  return (
    <div className="loader-container">
      <div className="loader" style={dynamicStyles}>Loading...</div>

      <style jsx>{`
        .loader-container {
          overflow: hidden;
        }

        .loader,
        .loader:after {
          border-radius: 50%;
        }

        .loader {
          margin: 0 auto;
          position: relative;
          text-indent: -9999em;

          -webkit-transform: translateZ(0);
          -ms-transform: translateZ(0);
          transform: translateZ(0);
          -webkit-animation: load8 0.8s infinite linear;
          animation: load8 0.8s infinite linear;
        }

        @-webkit-keyframes load8 {
          0% {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
          }
          100% {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
          }
        }

        @keyframes load8 {
          0% {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
          }
          100% {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
          }
        }
      `}</style>
    </div>
  );
}

export default Spinner;
