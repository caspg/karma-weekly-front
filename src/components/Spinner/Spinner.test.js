import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from './Spinner';

describe('Spinner component', () => {
  it('matches a snapshot with default props', () => {
    const component = shallow(
      <Spinner />
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('matches a snapshot with custom props', () => {
    const component = shallow(
      <Spinner
        mainColor="#000"
        bgColor="#fff"
        radius="50em"
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
