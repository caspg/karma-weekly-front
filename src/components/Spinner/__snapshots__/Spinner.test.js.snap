// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component matches a snapshot with custom props 1`] = `
<div
  className="loader-container"
  data-jsx={2137335714}
>
  <div
    className="loader"
    data-jsx={2137335714}
    style={
      Object {
        "borderBottom": "solid #fff",
        "borderLeft": "solid #000",
        "borderRight": "solid #fff",
        "borderTop": "solid #000",
        "borderWidth": "1em",
        "height": "50em",
        "width": "50em",
      }
    }
  >
    Loading...
  </div>
  <_class
    css=".loader-container[data-jsx=\\"2137335714\\"]{overflow:hidden}.loader[data-jsx=\\"2137335714\\"],.loader[data-jsx=\\"2137335714\\"]:after{border-radius:50%}.loader[data-jsx=\\"2137335714\\"]{margin:0 auto;position:relative;-webkit-text-indent:-9999em;text-indent:-9999em;-webkit-transform:translateZ(0);-ms-transform:translateZ(0);-webkit-transform:translateZ(0);-ms-transform:translateZ(0);transform:translateZ(0);-webkit-animation:load8 0.8s infinite linear;-webkit-animation:load8datajsx2137335714 0.8s infinite linear;animation:load8datajsx2137335714 0.8s infinite linear}0%[data-jsx=\\"2137335714\\"]{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%[data-jsx=\\"2137335714\\"]{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}@-webkit-keyframes load8datajsx2137335714{0%{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes load8datajsx2137335714{0%{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}}"
    styleId={2137335714}
  />
</div>
`;

exports[`Spinner component matches a snapshot with default props 1`] = `
<div
  className="loader-container"
  data-jsx={2137335714}
>
  <div
    className="loader"
    data-jsx={2137335714}
    style={
      Object {
        "borderBottom": "solid #ddd",
        "borderLeft": "solid #5dc0de",
        "borderRight": "solid #ddd",
        "borderTop": "solid #5dc0de",
        "borderWidth": "0.25em",
        "height": "10em",
        "width": "10em",
      }
    }
  >
    Loading...
  </div>
  <_class
    css=".loader-container[data-jsx=\\"2137335714\\"]{overflow:hidden}.loader[data-jsx=\\"2137335714\\"],.loader[data-jsx=\\"2137335714\\"]:after{border-radius:50%}.loader[data-jsx=\\"2137335714\\"]{margin:0 auto;position:relative;-webkit-text-indent:-9999em;text-indent:-9999em;-webkit-transform:translateZ(0);-ms-transform:translateZ(0);-webkit-transform:translateZ(0);-ms-transform:translateZ(0);transform:translateZ(0);-webkit-animation:load8 0.8s infinite linear;-webkit-animation:load8datajsx2137335714 0.8s infinite linear;animation:load8datajsx2137335714 0.8s infinite linear}0%[data-jsx=\\"2137335714\\"]{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%[data-jsx=\\"2137335714\\"]{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}@-webkit-keyframes load8datajsx2137335714{0%{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes load8datajsx2137335714{0%{-webkit-transform:rotate(0deg);-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(360deg);-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg)}}"
    styleId={2137335714}
  />
</div>
`;
